typescript.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+module.exports = {
+	extends: [
+		'plugin:@typescript-eslint/recommended'
+	],
+	plugins: [
+		'@typescript-eslint',
+		'@stylistic'
+	],
+	parser: '@typescript-eslint/parser',
+	parserOptions: {
+		sourceType: 'module',
+		ecmaFeatures: {
+			impliedStrict: true
+		}
+	},
+	rules: {
+		'comma-dangle': [ 'error', 'always-multiline' ],
+		// general note: rules not in the TS namespace act in combination with their following line
+		// align recommended w/ wikimedia style or escalating default warnings to errors
+		'@typescript-eslint/array-type': [ 'error', { default: 'array' } ],
+		'@typescript-eslint/explicit-function-return-type': [ 'error', {
+			allowExpressions: true,
+			allowTypedFunctionExpressions: true,
+			allowHigherOrderFunctions: true
+		} ],
+		'@typescript-eslint/explicit-module-boundary-types': [ 'error' ],
+		'@typescript-eslint/explicit-member-accessibility': [ 'error', { accessibility: 'explicit' } ],
+		// aligned to https://github.com/wikimedia/eslint-config-wikimedia/blob/master/common.json#L21
+		'@stylistic/indent': [ 'error', 'tab', { SwitchCase: 1 } ],
```

Comment:
We aren't using @stylistic in common yet so we should probably switch over at the same time for consistency.

You could also copy these over dynamically by calling `require( 'common' )` and selectively copying keys.